firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -133,8 +143,17 @@ public static FirebaseInstallations getInstance() {
    * @return a {@link FirebaseInstallations} instance
    */
   @NonNull
-  public static FirebaseInstallations getInstance(@NonNull FirebaseApp app) {
+  public static FirebaseInstallations getInstance(@NonNull FirebaseApp app)
+      throws FirebaseInstallationsException {
     Preconditions.checkArgument(app != null, "Null is not a valid value of FirebaseApp.");
+
+    if (!isIIDVersionCompatible()) {
+      throw new FirebaseInstallationsException(
+          "FirebaseInstallations will not work correctly with current version of Firebase "
+              + "Instance ID. Please update your Firebase Instance ID version.",
+          Status.INCOMPATIBLE_VERSIONS);
+    }
```

Comment:
Please outsource in dedicated method!

#preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk

Reply:
let the other method throw the exception!